core/src/main/java/org/apache/calcite/rel/core/JoinInfo.java

```diff
@@ -66,8 +66,36 @@ public static JoinInfo of(RelNode left, RelNode right, RexNode condition) {
     final List<RexNode> nonEquiList = new ArrayList<>();
     RelOptUtil.splitJoinCondition(left, right, condition, leftKeys, rightKeys,
         filterNulls, nonEquiList);
-    return new JoinInfo(ImmutableIntList.copyOf(leftKeys),
-        ImmutableIntList.copyOf(rightKeys), ImmutableList.copyOf(nonEquiList));
+
+    /**
+     * In join implementation when the condition is equality (=),
+     * we filter out nulls prior to the join. If the condition is
```

Comment:
There is no "we".
Moreover, this code has nothing to do with filtering.
This code is about building a description of the join predicate.
I think the JoinInfo class needs better documentation; in particular, it is not clear what the field nonEquiConditions contains. Perhaps some of this JavaDoc should move to document the fields themselves. Then we would not need to discuss about whether IS NOT DISTINCT FROM is an equiCondition or not.

Reply:
You are right, this description should be removed.